sdks/python/apache_beam/transforms/util.py

```diff
@@ -612,29 +611,23 @@ def restore_timestamps(element):
             for (value, timestamp) in values]
 
     else:
-      # The linter is confused.
-      # hash(1) is used to force "runtime" selection of _IdentityWindowFn
-      # pylint: disable=abstract-class-instantiated
-      cls = hash(1) and _IdentityWindowFn
-      window_fn = cls(
-          windowing_saved.windowfn.get_window_coder())
-
-      def reify_timestamps(element, timestamp=DoFn.TimestampParam):
+      def reify_timestamps(element,
+                           timestamp=DoFn.TimestampParam,
+                           window=DoFn.WindowParam):
         key, value = element
-        return key, TimestampedValue(value, timestamp)
+        # Transport the window as part of the value and restore it later.
+        return key, TimestampedValue((value, window), timestamp)
```

Comment:
Any reason not to use a WindowedValue here?

Reply:
Actually while making the change i realized that the timestamp will be duplicated when using the windowed values without any benefit. Hence dropping the use of windowed value.

Reply:
Why would the timestamp be duplicated? The (window, timestamp, value) tuple seems best represented by a WindowedValue. (I would be OK with a plain old 3-tuple as well, but only going half way seems odd).